lib/Conversion/ConvertToArcs/ConvertToArcs.cpp

```diff
@@ -278,7 +280,27 @@ void Converter::absorbRegs(HWModuleOp module) {
         isTrivial = false;
         break;
       }
+
       clock = regOp.getClk();
+      reset = regOp.getReset();
+
+      // Check that if the register has a reset, it is to a constant zero
+      if (reset) {
+        Value resetValue = regOp.getResetValue();
+        Operation *op = resetValue.getDefiningOp();
+        if (!op)
+          return regOp->emitOpError(
+              "is reset by an input; not supported by ConvertToArcs");
+        if (auto constant = dyn_cast<hw::ConstantOp>(op)) {
+          if (constant.getValue() != 0)
+            return regOp->emitOpError("is reset to a constant non-zero value; "
+                                      "not supported by ConvertToArcs");
+        } else {
+          return regOp->emitOpError("is reset to a value that is not clearly "
+                                    "constant; not supported by ConvertToArcs");
+        }
```

Comment:
Instead of erroring out here, we could represent the reset with a `comb.mux` that takes the reset condition as mux condition and the `resetValue` as `trueValue`. It would probably be easiest to do this in a pre-pass. But that's something we can add in a future PR. It's great that proper error messages are displayed now instead of just silently doing the wrong thing!

Reply:
Good point! Agreed that this would probably fit better in a future PR, but would definitely be nice to have!